Algorithms/NumberTheory/GreatestCommonDivisor.cpp

```diff
@@ -0,0 +1,65 @@
+//Name:Anuj Goyal
+//Email:anujg935@gmail.com
+
+//This program is to find out the Greatest common divisor of 2 numbers
+#include <iostream>
+using namespace std;
+
+//Naive algorithm for finding GCD
+//this loop starts from 2 to a+b
+//where a & b are 2 numbers
+int gcdNaive(int a, int b) {
+  int current_gcd = 1;
+  for (int d = 2; d <= a && d <= b; d++) {
+    if (a % d == 0 && b % d == 0) {
+      if (d > current_gcd) {
+        current_gcd = d;
+      }
+    }
+  }
+  return current_gcd;
+}
+
+//Euclid's algorithm for finding GCD

```

Comment:
Provide some explanation how this algorithm actually works!

Reply:
okay sure

Reply:
please tell me for which algo you want explanation naive or Euclid's

Reply:
Please provide an explanation for Euclid's algorithm.